javascript/javascript1/week3/homework5.js

```diff
@@ -0,0 +1,132 @@
+// CactusIO-interactive (Smart phone usage app) optional
+
+// Adding an activity
+
+const activities = [];
+/**
+ *
+ *
+ * @param {string} date
+ * @param {string} activity
+ * @param {number} duration
+ */
+const addActivity = (date, activity, duration) => {
+  activities.push({
+    date: date,
+    activity: activity,
+    duration: duration,
+  });
+};
+addActivity("23/7-18", "Youtube", 30);
+
+// Show my status
+
+// This function should use the activities variable and return
+// a string saying the following: "You have added 3 activities. They amount to 78 min. of usage"
+
+const usageLimit = 60;
+
+const showStatus = (arr) => {
+  let minutesOfUsage = 0;
+
+  for (let i = 0; i < arr.length; i++) {
+    minutesOfUsage += arr[i].duration;
+  }
+
+  if (minutesOfUsage >= usageLimit) {
+    return console.log(
+      "You have reached your limit, no more smartphoning for you!"
+    );
+  } else {
+    return console.log(
+      arr.length !== 0
+        ? `You have added ${arr.length} activities. They amount to ${minutesOfUsage} min. of usage`
+        : `Add some activities before calling showStatus`
+    );
+  }
+};
+console.log("\x1b[35m%s\x1b[0m", "\nShowstatus");
+showStatus(activities);
+
+// Usage limit
+
+// Improved addActivity, so that we dont need to specify the date,
+const date = new Date();
+const options = {
+  weekday: "long",
+  year: "numeric",
+  month: "long",
+  day: "numeric",
+};
+const dateToday = date.toLocaleDateString("en-EN", options);
+// function automatically figures out what the date is.
+const addActivityWithDate = (activity, duration) => {
+  activities.push({
```

Comment:
I would actually add the date initiation here. It's not wrong what you have done but it's related to the MVC (Data, Logic, Render).

If this was running on a server the date should be "called" then and there when you call the function. If it's outside the function you can't rely on the date as it could be when the server was deployed. Whereas if you put the date inside the function then it is called when the function is called.

I hope that makes sense, otherwise let me know!